EDM4eicVersion.h.in

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: Apache-2.0
+#ifndef EDM4EIC_VERSION_H
+#define EDM4EIC_VERSION_H
+
+#include <cstdint>
+#include <tuple>
+#include <ostream>
+#if __cplusplus >= 202002L
+#include <compare>
+#endif
+
+// Some preprocessor constants and macros for the use cases where they might be
+// necessary
+
+/// Define a version to be used in EDM4EIC.
+#define EDM4EIC_VERSION(major, minor, patch) ((UINT64_C(major) << 32) | (UINT64_C(minor) << 16) | (UINT64_C(patch)))
+/// Get the major version from a preprocessor defined version
+#define EDM4EIC_MAJOR_VERSION(v) (((v) & (-1UL >> 16)) >> 32)
+/// Get the minor version from a preprocessor defined version
+#define EDM4EIC_MINOR_VERSION(v) (((v) & (-1UL >> 32)) >> 16)
+/// Get the patch version from a preprocessor defined version
+#define EDM4EIC_PATCH_VERSION(v) ((v) & (-1UL >> 48))
+
+// Some helper constants that are populated by the cmake configure step
+#define EDM4EIC_VERSION_MAJOR @EDM4EIC_VERSION_MAJOR@
+#define EDM4EIC_VERSION_MINOR @EDM4EIC_VERSION_MINOR@
+#define EDM4EIC_VERSION_PATCH @EDM4EIC_VERSION_PATCH@
+#define EDM4EIC_VERSION EDM4EIC_VERSION(EDM4EIC_VERSION_MAJOR, EDM4EIC_VERSION_MINOR, EDM4EIC_VERSION_PATCH)
```

Comment:
Double definition?

Reply:
```suggestion
#define edm4hep_VERSION EDM4EIC_VERSION(EDM4EIC_VERSION_MAJOR, EDM4EIC_VERSION_MINOR, EDM4EIC_VERSION_PATCH)
```

Reply:
https://github.com/key4hep/EDM4hep/blob/main/EDM4hepVersion.h.in#L27C9-L27C24